homeassistant/components/unifiprotect/lock.py

```diff
@@ -0,0 +1,94 @@
+"""Support for Ubiquiti's UniFi Protect NVR."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyunifiprotect.data import Doorlock
+from pyunifiprotect.data.types import LockStatusType
+
+from homeassistant.components.lock import LockEntity, LockEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .data import ProtectData
+from .entity import ProtectDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Discover cameras with speakers on a UniFi Protect NVR."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            ProtectLock(
+                data,
+                lock,
+            )
+            for lock in data.api.bootstrap.doorlocks.values()
+        ]
```

Comment:
```suggestion
            ProtectLock(
                data,
                lock,
            )
            for lock in data.api.bootstrap.doorlocks.values()
```

This should work as a generator expression now that we require a new enough python